config/parse.go

```diff
@@ -291,7 +350,77 @@ func verifyItem(family string, name string, item interface{}) error {
 	}
 	err := i.Verify()
 	if err != nil {
+		log.WithFields(log.Fields{"family": family, "name": name}).Error("Invalid item configuration")
 		return skogul.Error{Source: "config parser", Reason: fmt.Sprintf("%s %s isn't valid", family, name), Next: err}
 	}
 	return nil
 }
+
+func findFieldsOfStruct(T reflect.Type) []string {
+	log.WithField("type", T.Name()).Debug("Finding defined fields for type")
+	requiredProps := make([]string, 0)
+	switch T.Kind() {
+	case reflect.Struct:
+		for i := 0; i < T.NumField(); i++ {
+			field := T.Field(i)
+			jsonTag := field.Tag.Get("json")
+
+			property := field.Name
+			if jsonTag != "" {
+				property = jsonTag
+			}
+			requiredProps = append(requiredProps, property)
+		}
+	}
+
+	return requiredProps
+}
+
+func getRelevantRawConfigSection(rawConfig *map[string]interface{}, family, section string) map[string]interface{} {
+	// log.Debugf("Fetching config section '%s' for '%s' from %v", section, family, rawConfig)
+	return (*rawConfig)[family].(map[string]interface{})[strings.ToLower(section)].(map[string]interface{})
+}
+
+func findMissingRequiredConfigProps(rawConfig *map[string]interface{}, family, handler string, T reflect.Type) []string {
```

Comment:
"handler" should be "section" to match getRelevantRawConfigSection - and because it isn't always a handler.

Reply:
Ok, having read this function: Doesn't this essentially make _all_ configurable variables required? Or am I missing something?

Reply:
That seems to be correct, yes. How do you think we should solve this? It could delegate configuration verification to the Verify() function and skip it completely here, only warning about superfluous parameters. Or warning about both, but not aborting the init/startup of skogul.

Reply:
I think Verify() is the only thing that knows whether something is required or not. I'm thinking we leave it up to that to complain about missing required configuration. Even warning would be misleading...